tests/e2e/component.go

```diff
@@ -61,6 +62,20 @@ func componentTests(componentCmdPrefix string) {
 		})
 	})
 
+	Context("Regression : listing component outside of component directory should fail", func() {
+		It("creates a component from local context, tries to list components from outside and fails", func() {
+			componentName := generateTimeBasedName("nodejs-ex")
+			dirName := generateTimeBasedName("context_dir")
+			runCmdShouldPass(fmt.Sprint("mkdir ", dirName))
+			runCmdShouldPass(componentCmdPrefix + " create nodejs " + componentName + " --git https://github.com/openshift/nodejs-ex --context " + dirName)
+			session := runCmdShouldFail("odo component list")
```

Comment:
this will not fail as the code above this line generates a `.odo` folder. this is the reason for the cryptic 
```
0
not equal to
0
```
in the tests

Reply:
Trying something different lets see

Reply:
ok I might be missing something but why does it not(never) fail

The line above takes `--context` but not component list

Reply:
Yea it is cryptic indeed. Anyway it should fail now. I am renaming odo directory before and renaming it back after